tests/test_toa_reader.py

```diff
@@ -132,3 +132,17 @@ def test_tttai():
     m = get_model(parstr)
     y = toa.get_TOAs("test1.tim", model=m)
     assert y.clock_corr_info["include_bipm"] == False
+
+
+def test_toa_merge():
+    filenames = ["NGC6440E.tim", "testtimes.tim", "parkes.toa"]
+    toas = [toa.get_TOAs(ff) for ff in filenames]
+    nt = toa.merge_TOAs(toas)
+    assert len(nt.observatories) == 3
+    assert nt.table.meta["filename"] == filenames
+    assert nt.ntoas == 78
+    # now test a failure if ephems are different
+    toas[0].ephem = "DE436"
+    nt = toa.merge_TOAs(toas)
+    assert nt is None
+
```

Comment:
It works be good to have tests that check whether this refuses illegal merges, and what happens if you successively merge multiple files one by one. You could also do a more exactly comparison between this and a TOAs merged using INCLUDE statements.

Reply:
I'm checking one type of the illegal merges.  But the syntax for all the others is exactly the same, so I'm not sure it is worth testing them all.  Consecutive merges now work and are tested.  Comparing closely with INCLUDEs is non-trivial given the way that commands are handled.  But INCLUDEs re handled better now (filenames of included files go in the .filename list) and they merge just fine (and that is tested).